File: classTypes4/java/src/Vaca.java
public class Vaca extends HerbivoroRumiante {
    private boolean daLeche;

    // constructor obligatorio
    public Vaca(double pesoKg, boolean daLeche) {
        // Llama al constructor de HerbivoroRumiante (que a su vez llama a Mamifero)
        super("Bos taurus", pesoKg); 
        this.daLeche = daLeche;
    }

    // metodos obligatorios implementados
    @Override
    public void emitirSonido() {
        System.out.println("moooo!");
    }
}

File: classTypes4/c#/Vaca.cs
public class Vaca : HerbivoroRumiante{
    private bool daLeche;

    // constructor obligatorio
    public Vaca(double pesoKg, bool daLeche) 
        // llama al constructor de herbivororumiante
        : base("bos taurus", pesoKg) 
    {
        this.daLeche = daLeche;
    }

    // metodos obligatorios implementados
    public override void emitirsonido(){
        Console.WriteLine("moooo!");
    }
}